src/js/grapl-cdk/lib/swarm.ts

```diff
@@ -49,115 +69,225 @@ export class Swarm extends cdk.Construct {
-         * https://github.com/docker/machine/issues/1655#issuecomment-409407523
-         *
-         * "DescribeSecurityGroups" -- required to check whether the
-         * --amazonec2-security-group actually exists
-         *
-         * "CreateSecurityGroup" -- not sure why this is required
-         *
-         * "DescribeSubnets" -- required to find the subnet
-         *
-         * "DescribeKeyPairs" -- to validate whether the keypair
-         * actually exists
-         *
-         * "CreateKeyPair" -- not sure why this is required
-         *
-         * these spot instance permissions apply if docker-machine is
-         * invoked with --amazonec2-request-spot-instance:
-         *
-         * "DescribeSpotInstances"
-         * "DescribeSpotInstanceRequests"
-         * "RequestSpotInstances"
-         * "CancelSpotInstanceRequests"
-         *
-         * "DescribeInstances" -- required to tell when an instance is
-         * ready, what its IP address is, etc
-         *
-         * "RunInstances" -- required to run an AWS instance if not
-         * --amazonec2-request-spot-instance
-         *
-         * "StartInstances" -- docker-machine start
-         *
-         * "StopInstances" -- docker-machine stop or docker-machine
-         * kill
-         *
-         * "RebootInstances" -- docker-machine restart
-         *
-         * "TerminateInstances" -- docker-machine rm
-         *
-         * "CreateTags" -- required to set the Name tag and anything
-         * that's passed via --amazonec2-tags
-         *
-         * "route53:ListHostedZonesByName" -- required to find the
-         * hosted zone ID
-         *
-         * "route53:ChangeResourceRecordSets" -- required to add A
-         * records to make DGraph discoverable.
-         */
-        const statement = new iam.PolicyStatement({
+        // CloudWatchAgentServerPolicy allows the Swarm instances to
+        // run the CloudWatch Agent.
+        swarmInstanceRole.addManagedPolicy(
+            iam.ManagedPolicy.fromAwsManagedPolicyName(
+                'CloudWatchAgentServerPolicy' // FIXME: don't use managed policy
+            )
+        );
+
+        // AmazonSSMManagedInstanceCore allows users to connect to
+        // instances with SSM
+        swarmInstanceRole.addManagedPolicy(
+            iam.ManagedPolicy.fromAwsManagedPolicyName(
+                'AmazonSSMManagedInstanceCore' // FIXME: don't use managed policy
+            )
+        );
+
+        // Logging policy to allow Swarm instances to ship service
+        // logs to CloudWatch.
+        swarmInstanceRole.addToPrincipalPolicy(new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: [
+                'logs:CreateLogGroup',
+                'logs:CreateLogStream',
+                'logs:PutLogEvents',
+                'logs:DescribeLogStreams',
+            ],
+            resources: [
+                'arn:aws:logs:*:*:*', // FIXME: lock this down more?
```

Comment:
Yeah, this looks like it could read logs from other deployments in the account.

I'm not sure what the logs arns look like, but hopefully the resource names begin with deployment name (or can) so we could limit to a deployment.

Reply:
Oh, I guess just overwrite, not read, so not as bad.

Reply:
I think I should be able to lock this down to a single log group's ARN.

Reply:
derp nvm this does things with log streams too.. still think I should be able to lock it down more, investigating.

Reply:
done